libs/upload/src/DragZone.tsx

```diff
@@ -291,13 +324,33 @@ function CustomUploadDropZone({
     className: `mx-auto ${tokens.iconColor}`,
   });
 
+  const [preLoadDelayPassed, setPreLoadDelayPassed] = useState(false);
+  const [postUploadPeriodActive, setPostUploadPeriodActive] = useState(false);
+
+  const prevUploadActive = usePrevious(uploadActive);
+
+  useEffect(() => {
+    if (uploadActive) {
+      setPreLoadDelayPassed(false);
+      setTimeout(() => {
+        setPreLoadDelayPassed(true);
+      }, preLoadDelay);
+      return () => void 0;
```

Comment:
I think you should clear appropriate timeout in the return, or leave it empty

Reply:
True, removed unnecessary returns